finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4DispatcherBase.scala

```diff
@@ -50,10 +53,9 @@ trait Netty4DispatcherBase[SendMsg <: Message, RecvMsg <: Message] {
     log.debug("[%s S:%d] initialized stream", prefix, id)
     val _ = stream.onReset.respond {
       case Return(_) =>
-        // Free and clear.
-        if (streams.replace(id, open, StreamClosed)) {
-          log.debug("[%s S:%d] stream closed", prefix, id)
-        }
+        closedId.updateAndGet(new IntUnaryOperator { def applyAsInt(i: Int) = if (id > i) id else i })
```

Comment:
might be able to avoid allocating a new IntUnaryOperator here by doing something like defining 

```
def addClosedId(id: Int): Unit = {
  val i = closedId.get
  val max = if (id > i) id else i
  if (!closedId.compareAndSet(i, max)) addClosedId(id()
}
```

which, I think, is pretty similar to what updateAndGet does anyway, based on the scaladoc.